src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/config/ResourceServerConfiguration.kt

```diff
@@ -21,12 +27,18 @@ class ResourceServerConfiguration {
         listOf(
           "/webjars/**", "/favicon.ico", "/csrf",
           "/health/**", "/info", "/startup", "/h2-console/**",
-          "/v3/api-docs/**", "/swagger-ui/**", "/swagger-ui.html",
+          "/v3/api-docs/**", "/swagger-ui.html",
           "/queue-admin/retry-all-dlqs",
           "/maintain-index/check-complete",
         ).forEach { authorize(it, permitAll) }
         authorize(anyExchange, authenticated)
       }
       oauth2ResourceServer { jwt { jwtAuthenticationConverter = AuthAwareTokenConverter() } }
     }
+
+  @Bean
+  fun locallyCachedJwtDecoder(
+    @Value("\${spring.security.oauth2.resourceserver.jwt.jwk-set-uri}") jwkSetUri: String,
+    cacheManager: CacheManager,
+  ): JwtDecoder = NimbusJwtDecoder.withJwkSetUri(jwkSetUri).cache(cacheManager.getCache("jwks")).build()
```

Comment:
note we haven't defined the cache elsewhere, so will just get a `ConcurrentHashMap` - https://docs.spring.io/spring-boot/docs/3.1.0/reference/htmlsingle/#io.caching.provider.simple